src/main/java/org/jenkinsci/plugins/workflow/graph/StandardGraphLookupView.java

```diff
@@ -147,6 +147,9 @@ BlockStartNode bruteForceScanForEnclosingBlock(@Nonnull final FlowNode node) {
             }
 
             // Now see if we have a winner among parents
+            if (current.getParents().isEmpty()) {
```

Comment:
I think we could make the inverse of this the while loop condition and stop special-casing `FlowStartNode`, but that would mean that we would load the parents of every node that we look at, even if they are `BlockEndNode`s that could skip straight to their `BlockStartNode`, so I opted for this local change.